Lib/test/test_capi/test_codecs.py

```diff
@@ -851,20 +852,29 @@ def test_codec_backslashreplace_errors_handler(self):
 
     def test_codec_namereplace_errors_handler(self):
         handler = _testlimitedcapi.codec_namereplace_errors
-        self.do_test_codec_errors_handler(handler, self.unicode_encode_errors)
+        self.do_test_codec_errors_handler(handler, self.unicode_encode_errors,
+                                          safe=True)
 
-    def do_test_codec_errors_handler(self, handler, exceptions):
+    def do_test_codec_errors_handler(self, handler, exceptions, *, safe=False):
         at_least_one = False
         for exc in exceptions:
             # See https://github.com/python/cpython/issues/123378 and related
             # discussion and issues for details.
-            if self._exception_may_crash(exc):
+            if not safe and self._exception_may_crash(exc):
                 continue
 
             at_least_one = True
             with self.subTest(handler=handler, exc=exc):
                 # test that the handler does not crash
-                self.assertIsInstance(handler(exc), tuple)
+                res = handler(exc)
+                self.assertIsInstance(res, tuple)
+                self.assertEqual(len(res), 2)
```

Comment:
Consider destructuring the tuple into two well-named variables.

Reply:
Nice catch. I can also now correctly check the range since the getters were now patched.